src/dimension/dimension_trait.rs

```diff
@@ -83,9 +83,27 @@ pub trait Dimension : Clone + Eq + Debug + Send + Sync + Default +
         self.slice().iter().fold(1, |s, &a| s * a as usize)
     }
 
-    /// Compute the size while checking for overflow.
+    /// Compute the size (number of elements) while checking for overflow.
+    ///
+    /// Returns `None` if the product of non-zero axis lengths overflows
+    /// `isize`.
+    ///
+    /// If `dim.size_checked()` returns `Some(size)`, a `Vec` is created of
+    /// length `size`, and `strides` are created with `self.default_strides()`
+    /// or `self.fortran_strides()`, then the invariants are met to construct
+    /// an owned `Array` from the `Vec`, `dim`, and `strides`. (See the docs of
+    /// `can_index_slice_not_custom` for a slightly more general case.)
     fn size_checked(&self) -> Option<usize> {
```

Comment:
I don't really see why this is needed, yet

Reply:
Ok, this seems good.

Isn't there an option to let this method just work with the overflow check and then check the isize::MAX requirement in constructors?

Reply:
Oh, whoops. I didn't realize that `size_checked` was a public method.

Initially, I didn't do this because I thought `size_checked` was private, and I noticed that all places but one where `.size_checked()` was called needed to perform the `isize::MAX` check too. (The only exception is [this line in `indices_iter_f`](https://github.com/rust-ndarray/ndarray/blob/master/src/indexes.rs#L215).)

I'll change it back to the original behavior and add a separate checking function for internal use.